src/Orleans.Runtime/Storage/StateStorageBridge.cs

```diff
@@ -48,17 +51,23 @@ public TState State
         /// <inheritdoc/>
         public bool RecordExists => GrainState.RecordExists;
 
-        public StateStorageBridge(string name, IGrainContext grainContext, IGrainStorage store, ILoggerFactory loggerFactory)
+        public StateStorageBridge(string name, IGrainContext grainContext, IGrainStorage store, ILoggerFactory loggerFactory, IActivatorProvider activatorProvider)
         {
             ArgumentNullException.ThrowIfNull(name);
             ArgumentNullException.ThrowIfNull(grainContext);
             ArgumentNullException.ThrowIfNull(store);
             ArgumentNullException.ThrowIfNull(loggerFactory);
+            ArgumentNullException.ThrowIfNull(activatorProvider);
 
             _logger = loggerFactory.CreateLogger(store.GetType());
             _name = name;
             _grainContext = grainContext;
             _store = store;
+
+            if (!typeof(TState).IsValueType)
+            {
+                _activator = activatorProvider.GetActivator<TState>();
+            }
```

Comment:
Currently the IActivatorProvider seems to not properly support value types, would it make more sense that this logic should be moved to it?